### DISCUSSION TITLE
How to run the k8ssandra Helm chart unit tests?

### DISCUSSION THREAD

**Discussion by carolili** [k8ssandra/k8ssandra#506] [Q&A] [answered]
Hello, could someone explain how to run the [unit tests](https://github.com/k8ssandra/k8ssandra/tree/main/tests/unit) for the Helm charts? They seem to be a new feature 🙂

I have tried running the unit tests for Prometheus as follows:
```
$ make TESTS=".*prometheus.*" unit-test
```

However, if changing [one of the values](https://github.com/k8ssandra/k8ssandra/blob/main/tests/unit/template_prometheus_test.go#L62) from `true` to `false`, the output from the command seems to still be the same.

```
$ make TESTS=".*prometheus.*" unit-test
go test  -test.timeout=3m ./tests/unit/... -coverprofile cover.out -args -ginkgo.focus=".*prometheus.*"
ok  	github.com/k8ssandra/k8ssandra/tests/unit	0.612s	coverage: [no statements]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/cassdc	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/helm	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/kubeapi	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/traefik	[no test files]
```
So I'm wondering if the changes are not taken into account or if I am running the unit tests in the wrong way. I also tried checking the documentation but could not find any details. Many thanks in advance.

**Comment by jsanda** [comment, marked as answer]
Hi @carolili 

You hit #507. I just submitted a PR for it. Note that even with the fix `TESTS=".*prometheus.*"` won't match any tests. We do not have a lot of unit tests around Prometheus since most of the Prometheus integration is handled by kube-prometheus-stack.

>I also tried checking the documentation but could not find any details.

Developer docs are coming 😄

**Reply by jsanda** [reply]
Since #507 turned out to be a non-issue, I think the problem you ran into is due to the fact that there are no tests matching the `.*prometheus.*` regex.

**Comment by carolili** [comment]
Ok, that makes sense. Thank you for the investigation and response in #507 !
